packages/opentelemetry-plugin-user-interaction/test/userInteraction.test.ts

```diff
@@ -0,0 +1,393 @@
+        called = true;
+      };
+      fakeInteraction(callback, btn);
+      sandbox.clock.tick(1000);
+      originalSetTimeout(() => {
+        assert.equal(called, false, 'callback should not be called');
+        done();
+      });
+    });
+
+    it('should handle 3 overlapping interactions', done => {
+      const btn1 = document.createElement('button');
+      btn1.setAttribute('id', 'btn1');
+      const btn2 = document.createElement('button');
+      btn2.setAttribute('id', 'btn2');
+      const btn3 = document.createElement('button');
+      btn3.setAttribute('id', 'btn3');
+      fakeInteraction(() => {
+        getData(FILE_URL, () => {
+          sandbox.clock.tick(10);
+        }).then(() => {});
+      }, btn1);
+      fakeInteraction(() => {
+        getData(FILE_URL, () => {
+          sandbox.clock.tick(10);
+        }).then(() => {});
+      }, btn2);
+      fakeInteraction(() => {
+        getData(FILE_URL, () => {
+          sandbox.clock.tick(10);
+        }).then(() => {});
+      }, btn3);
+      sandbox.clock.tick(1000);
+      originalSetTimeout(() => {
+        assert.equal(exportSpy.args.length, 6, 'should export 6 spans');
+
+        const span1: tracing.ReadableSpan = exportSpy.args[0][0][0];
+        const span2: tracing.ReadableSpan = exportSpy.args[1][0][0];
+        const span3: tracing.ReadableSpan = exportSpy.args[2][0][0];
+        const span4: tracing.ReadableSpan = exportSpy.args[3][0][0];
+        const span5: tracing.ReadableSpan = exportSpy.args[4][0][0];
+        const span6: tracing.ReadableSpan = exportSpy.args[5][0][0];
+
+        assertClickSpan(span1, 'btn1');
+        assertClickSpan(span2, 'btn2');
+        assertClickSpan(span3, 'btn3');
+
+        assert.strictEqual(
+          span1.spanContext.spanId,
+          span4.parentSpanId,
+          'span4 has wrong parent'
+        );
+        assert.strictEqual(
+          span2.spanContext.spanId,
+          span5.parentSpanId,
+          'span5 has wrong parent'
+        );
+        assert.strictEqual(
+          span3.spanContext.spanId,
+          span6.parentSpanId,
+          'span6 has wrong parent'
+        );
+
+        done();
+      });
+    });
+
+    describe('_createSpan', () => {
+      let createSpan: any;
+      beforeEach(() => {
+        createSpan = userInteractionPlugin['_createSpan'].bind(
+          userInteractionPlugin
+        );
+      });
+
+      it('it should not create span when element has no getAttribute', () => {
+        const span = createSpan({} as any, '');
+        assert.strictEqual(span, undefined, 'span should not be created');
+      });
+
+      it('it should not create span when element is disabled', () => {
+        const btn = createButton(true);
+        const span = createSpan(btn, '');
+        assert.strictEqual(span, undefined, 'span should not be created');
+      });
+
+      it('it should not create span when error occurred during span creation', () => {
+        // @ts-ignore
```

Comment:
Why is `ts-ignore` needed here? Is there a way to refactor the code to not need it?

Reply:
there is no such possibility, tracer is private, this is just the unit test and purpose is to simulate the failure in adding the span in tracer

Reply:
You can access private properties by doing `userInteractionPlugin["_tracer"].startSpan`